### PR TITLE
Zeekling dives kinda parabolically and screams

### PR DESCRIPTION
**Pull request by erratic-c-programmer**
This may fix #784.
The 'parabola' is just increasing acceleration per frame.

### CODE REVIEW

**Comment by tobbi** [comment]
Hey,

there are a couple compiler warnings with this:
```
/home/travis/build/SuperTux/supertux/src/badguy/zeekling.cpp: In constructor ‘Zeekling::Zeekling(const ReaderMapping&)’:

/home/travis/build/SuperTux/supertux/src/badguy/zeekling.cpp:27:1: error: ‘Zeekling::dive_so_far’ should be initialized in the member initialization list [-Werror=effc++]

 Zeekling::Zeekling(const ReaderMapping& reader) :

 ^~~~~~~~

/home/travis/build/SuperTux/supertux/src/badguy/zeekling.cpp: In member function ‘virtual void Zeekling::active_update(float)’:

/home/travis/build/SuperTux/supertux/src/badguy/zeekling.cpp:178:21: error: conversion from ‘double’ to ‘float’ changes value from ‘5.0000000000000003e-2’ to ‘5.00000007e-2f’ [-Werror=float-conversion]

       dive_so_far = 0.05;

                     ^~~~

/home/travis/build/SuperTux/supertux/src/badguy/zeekling.cpp:186:17: error: conversion from ‘double’ to ‘float’ may change value [-Werror=float-conversion]

     dive_so_far += 0.1;
```

Please fix them so this can get merged.

File: src/badguy/zeekling.cpp [hunk 1]
```diff
@@ -22,6 +22,7 @@
 #include "math/random.hpp"
 #include "object/player.hpp"
 #include "sprite/sprite.hpp"
+#include "audio/sound_manager.hpp"
```
Comment (weluvgoatz) [inline src/badguy/zeekling.cpp:25]: Please put the includes in alphabetical order, not counting the one referencing the hpp file.  Here, you just slapped it on at the end when it should really go not there.

File: src/badguy/zeekling.cpp [hunk 2]
```diff
@@ -176,15 +175,20 @@ Zeekling::active_update(float dt_sec) {
...
-      m_physic.set_velocity_y(2*fabsf(m_physic.get_velocity_x()));
+      dive_so_far = 0.05;
+      m_physic.set_velocity_y(dive_so_far * fabsf(m_physic.get_velocity_x()));
       m_sprite->set_action(m_dir == Direction::LEFT ? "diving-left" : "diving-right");
+      SoundManager::current()->play("sounds/deathd.wav");
     }
     BadGuy::active_update(dt_sec);
     return;
   } else if (state == DIVING) {
+    dive_so_far += 0.1;
```
Comment (weluvgoatz) [inline src/badguy/zeekling.cpp:186]: The way you did the curve makes the diving look quite terrible and not as impactful, seen as the way it hits Tux also is affected.

**Comment by erratic-c-programmer** [comment]
Well, I don't really know how to go about doing this besides that rather 'gravity-like' curve, so we'll just have to wait for someone who actually knows how to do this to come along...

**Pull request closed without merging**